cirq/ops/pauli_string.py

```diff
@@ -187,6 +190,25 @@ def __str__(self):
 
         return prefix + '*'.join(factors)
 
+    def _has_unitary_(self) -> bool:
```

Comment:
aside: does this point out we need another protocol for general objects that have a matrix but are not unitary

Reply:
Possibly. We could certainly add a `matrix` method to this class. In general I would like to have the protocols be driven by "I need that to be available to do X" rather than "I have a Y available and could provide it generically".